### PR TITLE
Ensure a metadata ID is at least a tick old before moving the summary pointer to it [BA-6159]

### PR DESCRIPTION
**Pull request by cjllanwarne**
A "not-perfect but better than doing nothing" way to fix the summarization issue.

Solution 5 from: https://docs.google.com/presentation/d/1amxelFufCIwwk43Zcufxn8FEWdS1jXb1d-oTaq9A0TU/edit?usp=sharing

### CODE REVIEW

File: database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala [hunk 1]
```diff
@@ -78,10 +78,11 @@ trait MetadataSqlDatabase extends SqlDatabase {
                           rootWorkflowIdKey: String,
                           labelMetadataKey: String,
                           limit: Int,
+                          nextSummaryPointerUpdateCeiling: Option[Long],
```
Comment (mcovarr) [inline database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala]: scaladoc

File: database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala [hunk 2]
```diff
@@ -78,10 +78,11 @@ trait MetadataSqlDatabase extends SqlDatabase {
...
                           buildUpdatedSummary:
                           (Option[WorkflowMetadataSummaryEntry], Seq[MetadataEntry])
                             => WorkflowMetadataSummaryEntry)
-                         (implicit ec: ExecutionContext): Future[(Long, Long)]
+                         (implicit ec: ExecutionContext): Future[(Long, Long, Long)]
```
Comment (mcovarr) [inline database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala]: TOL I guess but that truple really ought to be a type

File: services/src/main/scala/cromwell/services/metadata/impl/MetadataDatabaseAccess.scala [hunk 3]
```diff
@@ -178,6 +178,7 @@ trait MetadataDatabaseAccess {
         rootWorkflowIdKey = WorkflowMetadataKeys.RootWorkflowId,
         labelMetadataKey = WorkflowMetadataKeys.Labels,
         limit = limit,
+        nextSummaryPointerUpdateCeiling: Option[Long],
```
Comment (mcovarr) [inline services/src/main/scala/cromwell/services/metadata/impl/MetadataDatabaseAccess.scala]: weird and different from all the others

File: services/src/main/scala/cromwell/services/metadata/impl/MetadataSummaryRefreshActor.scala [hunk 4]
```diff
@@ -29,13 +29,17 @@ object MetadataSummaryRefreshActor {
   sealed trait SummaryRefreshState
   case object WaitingForRequest extends SummaryRefreshState
   case object SummarizingMetadata extends SummaryRefreshState
-  case object MetadataSummaryComplete extends SummaryRefreshState
 
-  case object SummaryRefreshData
+  sealed trait SummaryRefreshData
+  case object EmptySummaryRefreshData extends SummaryRefreshData
+  final case class KnownNextSummaryStatusPointerCeiling(ceiling: Long) extends SummaryRefreshData
```
Comment (mcovarr) [inline services/src/main/scala/cromwell/services/metadata/impl/MetadataSummaryRefreshActor.scala]: standardize terminology plz: `KnownNextSummaryStatusPointerCeiling` vs `nextSummaryPointerUpdateCeiling`

File: database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala (hunk 2 above)
Comment (cjllanwarne) [reply database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala]: I _think_ @kshakir scared me off making things in `cromwell/database/` into types... but maybe I over-corrected?

File: services/src/main/scala/cromwell/services/metadata/impl/MetadataSummaryRefreshActor.scala (hunk 4 above)
Comment (cjllanwarne) [reply services/src/main/scala/cromwell/services/metadata/impl/MetadataSummaryRefreshActor.scala]: This is a `StateData` type for an actor, so this represents that the actor has a "known" `nextSummaryPointerUpdateCeiling`. I'll see if I can come up with a better naming scheme

File: database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala (hunk 1 above)
Comment (cjllanwarne) [reply database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala]: Added. And for your reference: `@param nextSummaryPointerUpdateCeiling If provided, the highest value to which the summary status pointer can be updated. If not set, the pointer will not move.`

File: database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala (hunk 2 above)
Comment (mcovarr) [reply database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala]: Yeah, you may be right. We can always come back to this later if that's not the case.

**Comment by mcovarr** [comment]
Four broken sub-builds 😬

**Comment by mcovarr** [comment]
I was having trouble with my PR builds so I blew away all the caches; that seems to have had beneficial effects here too. 🙂

**Pull request merged**